int32[] correct_indexes
---
bool success
